aderyn_driver/src/process_auto.rs

```diff
@@ -162,7 +162,16 @@ fn absorb_ast_content_into_context(
     root_path: &Path,
     context: &mut WorkspaceContext,
 ) {
-    let mut source_unit: SourceUnit = serde_json::from_str(ast_content).unwrap();
+    let source_unit_deserializer = &mut serde_json::Deserializer::from_str(ast_content);
+    let result: Result<SourceUnit, _> = serde_path_to_error::deserialize(source_unit_deserializer);
+    let mut source_unit: SourceUnit = match result {
+        Ok(source_unit) => source_unit,
+        Err(err) => {
+            let path = err.path().to_string();
+            println!("Faieled to deserialize at path {:?}", path);
+            std::process::exit(-1);
+        }
+    };
```

Comment:
2nd thoughts .. before merging. Deserializing this way could be very slow and I think it should only be used in tests.

Reply:
For now let us only use this branch for debugging purposes. I think once the AST is complete, we wouldn't need to worry too much